src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -139,11 +139,15 @@ public IndexableRecord convertRecord(
         throw new ConnectException("routing field should be primitive or String");
       }
     }
+    ActionableRecord.ActionType actionType = ActionableRecord.ActionType.Index;
+    if (value instanceof HashMap && ((HashMap) value).get("doc") != null) {
```

Comment:
Is there a particular reason why you are checking for `HashMap` specifically vs checking for instance of `Map`?

Reply:
Are we guaranteed that it would not change to something like a `ConcurrentHashMap` in the implementation?

Reply:
When we change that in data-pipelines project (e.g to maybe a Struct), we'll change this.